src/components/App.js

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState } from "react";
+import LocationDetails from "./LocationDetails";
+import ForecastSummaries from "./ForecastSummaries";
+import ForecastDetails from "./ForecastDetails";
+import getForecast from "../requests/getForecast";
+import SearchForm from "./SearchForm";
+import "../styles/App.css";
+
+const App = () => {
+  const [forecasts, setForecasts] = useState([]);
+  const [location, setLocation] = useState({ city: "", country: "" });
+  const [selectedDate, setSelectedDate] = useState(0);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [searchText, setSearchText] = useState("");
+  const selectedForecast = forecasts.find(
+    (forecast) => forecast.date === selectedDate
+  );
+  const handleForecastSelect = (date) => {
+    setSelectedDate(date);
+  };
+  const handleCitySearch = () => {
+    getForecast(
+      searchText,
+      setSelectedDate,
+      setForecasts,
+      setLocation,
+      setErrorMessage
+    );
+  };
+
+  useEffect(() => {
+    getForecast(
+      searchText,
+      setSelectedDate,
+      setForecasts,
+      setLocation,
+      setErrorMessage
+    );
+  }, []);
+
+  return (
+    <div className="weather-app">
+      <header>
+        <h1>Its always cloudy in {location.city}.</h1>
+      </header>
+      <br />
+      <LocationDetails
+        city={location.city}
+        country={location.country}
+        errorMessage={errorMessage}
+      />
+      <br />
```

Comment:
`<br>`s really shouldnt be used anymore, use CSS to create margins between elements

Reply:
Removed.